resolwe_bio/processes/samtools/samtools_bedcov.py

```diff
@@ -0,0 +1,167 @@
+"""Samtools bedcov."""
+
+from io import StringIO
+from pathlib import Path
+
+import pandas as pd
+from plumbum import TEE
+
+from resolwe.process import (
+    BooleanField,
+    Cmd,
+    DataField,
+    FileField,
+    GroupField,
+    IntegerField,
+    Process,
+    SchedulingClass,
+    StringField,
+)
+
+
+def normalize_bedcov_output(output: str) -> pd.DataFrame:
+    """Calculate mean coverage over regions specified in samtools bedcov output.
+
+    Samtools bedcov returns sum of read coverages across all bases in bed file regions.
+    This function normalizes the values with the length of individual regions.
+
+    Input columns : [chr, start, stop, coverage sum]
+    Output columns : [chr, start, stop, coverage mean]
+    """
+    coverage_df = pd.read_csv(
+        StringIO(output),
+        sep="\t",
+        names=["chrom", "start", "end", "coverage"],
+    )
+
+    coverage_df["coverage"] = coverage_df["coverage"] / (
+        coverage_df["end"] - coverage_df["start"]
+    )
+
+    return coverage_df
+
+
+class SamtoolsBedcov(Process):
+    """Samtools bedcov.
+
+    Reports the total read base count (i.e. the sum of per base read depths)
+    for each genomic region specified in the supplied BED file.
+    The regions are output as they appear in the BED file and are 0-based.
+    The output is tab-delimited with the first three columns being the chromosome,
+    start and end positions of the region, and the remaining column
+    being the sum read base counts or normalized sum read base counts by the length of
+    individual regions (mean coverage).
+
+    For more information about samtools bedcov, click
+    [here](https://www.htslib.org/doc/samtools-bedcov.html).
+
+    """
+
+    slug = "samtools-bedcov"
+    process_type = "data:alignment:bam:samtools"
+    name = "Samtools bedcov"
+    requirements = {
+        "expression-engine": "jinja",
+        "executor": {
+            "docker": {"image": "public.ecr.aws/s4q6j6e8/resolwebio/common:3.1.0"},
+        },
+        "resources": {
+            "cores": 2,
+            "memory": 16384,
```

Comment:
I suspect the process to be fairly efficient. We might get away with less resources. I suggest we drop this to 8GB RAM.